microsoftdynamicscrm/metadata_test.go

```diff
@@ -0,0 +1,217 @@
+				w.WriteHeader(http.StatusOK)
+				writeBody(w, `<?xml version="1.0" encoding="utf-8"?>`)
+			})),
+			expectedErrs: []error{common.ErrNoXMLRoot},
+		},
+		{
+			name:  "Server response without Sales Schema",
+			input: []string{"msfp_surveyinvite"},
+			server: httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("Content-Type", "application/xml")
+				w.WriteHeader(http.StatusOK)
+				writeBody(w, `
+<?xml version="1.0" encoding="utf-8"?>
+<edmx:Edmx Version="4.0" xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx"></edmx:Edmx>`)
+			})),
+			expectedErrs: []error{ErrMissingSchema},
+		},
+		{
+			name:  "Object name cannot be found from server response",
+			input: []string{"msfp_surveyinvite"},
+			server: httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("Content-Type", "application/xml")
+				w.WriteHeader(http.StatusOK)
+				_, _ = w.Write(fakeServerResp)
+			})),
+			expectedErrs: []error{ErrObjectNotFound, errors.New("unknown entity msfp_surveyinvite")}, // nolint:goerr113
+		},
+		{
+			name:  "Correctly list metadata for account leads and invite contact",
+			input: []string{"accountleads", "adx_invitation_invitecontacts"},
+			server: httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("Content-Type", "application/xml")
+				w.WriteHeader(http.StatusOK)
+				_, _ = w.Write(fakeServerResp)
+			})),
+			expected: &common.ListObjectMetadataResult{
+				Result: map[string]common.ObjectMetadata{
+					"accountleads": {
+						DisplayName: "accountleads",
+						FieldsMap: map[string]string{
+							"accountid":                 "accountid",
+							"accountleadid":             "accountleadid",
+							"importsequencenumber":      "importsequencenumber",
+							"leadid":                    "leadid",
+							"name":                      "name",
+							"overriddencreatedon":       "overriddencreatedon",
+							"timezoneruleversionnumber": "timezoneruleversionnumber",
+							"utcconversiontimezonecode": "utcconversiontimezonecode",
+							"versionnumber":             "versionnumber",
+						},
+					},
+					"adx_invitation_invitecontacts": {
+						DisplayName: "adx_invitation_invitecontacts",
+						FieldsMap: map[string]string{
+							"adx_invitation_invitecontactsid": "adx_invitation_invitecontactsid",
+							"adx_invitationid":                "adx_invitationid",
+							"contactid":                       "contactid",
+							"versionnumber":                   "versionnumber",
+						},
+					},
+				},
+				Errors: nil,
+			},
+			expectedErrs: nil,
+		},
+		{
+			// In total phonecall will have 65 fields, where
+			// phonecall 		(has 7 fields) and inherits from
+			// activitypointer 	(has 58 fields), which in turn inherits from
+			// crmbaseentity 	(has 0 fields)
+			name:  "Correctly list metadata for phone calls including inherited fields",
```

Comment:
Real schemas are huge. Even the small one selected for the test will make response check cumbersome. For this case we just check that total number of fields are indeed inherited and match expectation.